### ISSUE TITLE
📎 `lint/noUnusedVariables` should ignore unused spread rest siblings

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#1910]
### Discussed in https://github.com/biomejs/biome/discussions/572

<div type='discussions-op-text'>

<sup>Originally posted by **EvHaus** October 21, 2023</sup>
A common pattern to omit keys from an object is to spread it like so:

```js
const car = {model: "Model S", brand: "Tesla", year: 2019};

// Remove model & year keys from the object
const {model, year, ...other} = car;

// Use the new object
formatBrand(other);
```

This is effectively equivalent to:

```js
const car = {model: "Model S", brand: "Tesla", year: 2019};

delete car.model;
delete car.year;

// Use the new object
formatBrand(car);
```

Except that it's an immutable way of doing it, and doesn't violate the `performance.noDelete` rule.

Unfortunately, the `correctness/noUnusedVariables` rule currently (and correctly) flags this syntax as a violation, as the `model` and `year` variables now become unused.

---

Thus, we should allow unused rest siblings. I think we don't need any option for this. Note that an unused spread should still be reported.
